packages/guides-markdown/src/Markdown/Parsers/BlockQuoteParser.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Markdown\Parsers;
+
+use League\CommonMark\Extension\CommonMark\Node\Block\BlockQuote;
+use League\CommonMark\Node\Node as CommonMarkNode;
+use League\CommonMark\Node\NodeWalker;
+use League\CommonMark\Node\NodeWalkerEvent;
+use phpDocumentor\Guides\MarkupLanguageParser;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\QuoteNode;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+
+use function sprintf;
+
+/** @extends AbstractBlockParser<QuoteNode> */
+final class BlockQuoteParser extends AbstractBlockParser
+{
+    /** @param iterable<AbstractBlockParser<Node>> $subParsers */
+    public function __construct(
+        private readonly iterable $subParsers,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    public function parse(MarkupLanguageParser $parser, NodeWalker $walker, CommonMarkNode $current): QuoteNode
+    {
+        $content = [];
+
+        while ($event = $walker->next()) {
+            $commonMarkNode = $event->getNode();
+
+            if ($event->isEntering()) {
+                foreach ($this->subParsers as $subParser) {
+                    if ($subParser->supports($event)) {
+                        $content[] = $subParser->parse($parser, $walker, $commonMarkNode);
+                        break;
+                    }
+                }
+
+                continue;
+            }
+
+            // leaving the heading node
+            if ($commonMarkNode instanceof BlockQuote) {
+                return new QuoteNode($content);
+            }
+
+            $this->logger->warning(sprintf('BlockQuote CONTEXT: I am leaving a %s node', $commonMarkNode::class));
```

Comment:
```suggestion
            $this->logger->debug(sprintf('BlockQuote CONTEXT: I am leaving a %s node', $commonMarkNode::class));
```

Reply:
I think we should change the message not the logging type. When this happens it means that the commonmarks node was not expected